internal/pkg/agent/application/upgrade/step_unpack.go

```diff
@@ -351,6 +378,46 @@ func untar(log *logger.Logger, version string, archivePath, dataDir string) (Unp
 	}, nil
 }
 
+func getPackageMetadataFromTar(archivePath string) (packageMetadata, error) {
+	// quickly open the archive and look up manifest.yaml file
+	fileContents, err := getFilesContentFromTar(archivePath, "manifest.yaml", agentCommitFile)
```

Comment:
better to have `manifest.yaml` in a const

Reply:
Done already as part of #4178 https://github.com/elastic/elastic-agent/pull/4178/commits/8bbd55f8820e29440db1b80a6c757fcb45ee5a7b